src/cascadia/WindowsTerminal/AppHost.cpp

```diff
@@ -669,23 +669,27 @@ void AppHost::_listenForInboundConnections()
 winrt::fire_and_forget AppHost::_setupGlobalHotkeys()
 {
     // The hotkey MUST be registered on the main thread. It will fail otherwise!
-    co_await winrt::resume_foreground(_logic.GetRoot().Dispatcher(),
-                                      winrt::Windows::UI::Core::CoreDispatcherPriority::Normal);
+    co_await winrt::resume_foreground(_logic.GetRoot().Dispatcher());
 
-    // Remove all the already registered hotkeys before setting up the new ones.
-    _window->UnsetHotkeys(_hotkeys);
+    // Unregister all previously registered hotkeys.
+    //
+    // RegisterHotKey(), will not unregister hotkeys automatically.
+    // If a hotkey with a given HWND and ID combination already exists
+    // then a duplicate one will be added, which we don't want.
+    // (Additionally we want to remove hotkeys that were removed from the settings.)
+    _window->UnregisterAllHotKeys();
 
-    _hotkeyActions = _logic.GlobalHotkeys();
     _hotkeys.clear();
-    for (const auto& [k, v] : _hotkeyActions)
+
+    // Re-register all current hotkeys.
+    for (const auto& [keyChord, cmd] : _logic.GlobalHotkeys())
     {
-        if (k != nullptr)
+        if (auto summonArgs = cmd.ActionAndArgs().Args().try_as<Settings::Model::GlobalSummonArgs>())

```

Comment:
By casting the `cmd` prematurely, we won't have to cast it again in `AppHost::_GlobalHotkeyPressed`.
Additionally we won't have to hold onto the `_hotkeyActions` map.

I explain this change a bit more in `IslandWindow::RegisterHotKey` below.